Feature/Oauth public and Marketplace update (#1309)
mercadopago/devsite-docs: guides/online-payments/marketplace/checkout-api/create-marketplace.en.md

Lastly, you need to get your application ID in [Your Integrations](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/panel).

## Link your Mercado Pago account with your sellers' accounts

To operate in Mercado Pago on behalf of your seller, you need to request their authorization first. To manage several Mercado Pago accounts at the same time in your integration, you can do it through OAuth, a feature for secure linking that allows sellers to access their Mercado Pago account to authorize and enable your application to work under their name.

To do this, you need to include a URL in your application to redirect sellers to the authorization site. 

This is the example URL you can use. You will also find the details of parameters to fill it out.

```url
https://auth.mercadopago[FAKER][URL][DOMAIN]/authorization?client_id=APP_ID&response_type=code&platform_id=mp&state=RANDOM_ID&redirect_uri=https://www.redirect-url.com

```
| Parameter | Data to fill out |
| ----------------- | ----------------- |
| `client_id` | Replace `APP_ID` value with your application ID. |

> | `state` | Identify who the code to be received belongs to; so replace `RANDOM_ID` value with a unique ID for each attempt without sensitive data. |

Reviewer: ```suggestion
| `state` | Identify who the code to be received belongs to. In order to do this, replace `RANDOM_ID` value with a unique ID for each attempt without sensitive data. |
```

| `redirect_uri` | Add the URL you entered in the Redirect URL field when you set up your application. | 

> When entering this URL, the seller will be redirected to Mercado Pago to log in their account and authorize the link to your application.

Reviewer: ```suggestion
When entering this URL, the seller will be redirected to Mercado Pago to log into their account and authorize the link to your application.
```

     -d 'client_secret=CLIENT_SECRET' \
     -d 'client_ID=CLIENT_ID' \
     -d 'grant_type=authorization_code' \
     -d 'code=CODE' \
     -d 'redirect_uri=REDIRECT_URI'
```

In the response, you will get the `access_token` of the linked seller. 
You will also get a `refresh_token` to be later used to refresh your seller's credentials. 

You will also receive the seller's `public_key`, which is the credential or public key to be used for your frontend account identification. 

```json
{
"access_token":"APP_USR-4934588586838432-XXXXXXXX-241983636",
"token_type": "bearer",
"expires_in": 15552000,
"scope": "offline_access read write",
"user_id": 241983636,
"refresh_token": "TG-XXXXXXXX-241983636",
"public_key": "APP_USR-d0a26210-XXXXXXXX-479f0400869e",
"live_mode": true
}
```

> WARNING 
> 
> Important
> 
> Remember that you will be using your seller's sensitive information. Secure a backup and do not include it in your link URLs; manage such information from your server only.

Done! You already linked your seller's account to your application via OAuth. 

> Keep in mind that these steps need to be repeated with each account you want to link. 

## Refresh your credentials

> **Your sellers' data that you received last 180 days**. After that, you need to request the seller's authorization again.

Reviewer: ```suggestion
**Your sellers' data that you received lasts 180 days**. After that, you need to request the seller's authorization again.
```

> **Your sellers' data that you received last 180 days**. After that, you need to request the seller's authorization again.

Author: ```suggestion
**Your sellers' data that you received lasts 180 days**. After that, you need to request the seller's authorization again.
```

To avoid this, renew data before the deadline to ensure they are always current. 

To refresh them, use the following OAuth API call:

```curl
curl -X POST \
     -H 'accept: application/json' \
     -H 'content-type: application/x-www-form-urlencoded' \
     'https://api.mercadopago.com/oauth/token' \
     -d 'client_secret=CLIENT_SECRET' \
     -d 'client_id=CLIENT_ID' \
     -d 'grant_type=refresh_token' \
     -d 'refresh_token=USER_REFRESH_TOKEN'
```

| Parameter | Description |
| ----------------- | ----------------- |
| `client_secret` | Use your `client_secret` key. |
| `client_id` | Use your `client_id` credential. |
| `grant_type` | Include `refresh_token` that remains unchanged. |
| `refresh_token` | Value received with your seller's data. | 

You will receive the following response:

```json
{
    "access_token": "APP_USR-4934588586838432-XXXXXXXX-241983636",
    "token_type": "bearer",
    "expires_in": 15552000,
    "scope": "offline_access read write",
    "refresh_token": "TG-XXXXXXXXXXXX-241983636"
}
```

> NOTE
> 
> Note
> 
> Remember that every time you refresh your credentials, the `refresh_token` will also change so you will need to store it again.
>

> > In case of errors when refreshing your credentials, remember that you can query them in the [error code reference](https://developers.mercadolibre[FAKER][URL][DOMAIN]/en_us/authentication-and-authorization#Error-codes-reference).

Reviewer: ```suggestion
>  In case of errors when refreshing your credentials, remember that you can query them in the [error code reference](https://developers.mercadolibre[FAKER][URL][DOMAIN]/en_us/authentication-and-authorization#Error-codes-reference).
```

## Test your integration

You can try your Marketplace using your Sandbox credentials to associate the sellers and to make the payments/refunds/cancellations.  
[Test your integration](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/online-payments/checkout-api/testing)

## Set up notifications

You can get notifications every time a seller links to or unlinks from your application. For setup, follow these steps:

1. Access [Your Applications](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/panel) and select the application used for OAuth flow.

2. Go to "Webhooks Notifications" tab. In this section, go to "Production Mode" and add the URL to receive notifications. If you want, you can click on "Test" button to test that the assigned URL will receive Webhooks Notifications correctly.

3. Then, in "Events" field, select "Application Linking" option. Lastly, click on save. 

Done! Every time a seller links or unlinks, a notification will be received in your assigned URL.

> These are some of the data that you can find within the notifications:

Reviewer: ```suggestion
This is some of the data that you can find within the notifications:
```

| Attribute | Value or type | Description |
| ----------------- | ----------------- | --------------- |
| `type` | `mp-connect` | Identifies the notification of account link type. |

> | `action` | `application.authorized` | Reports that seller linked to application. |

Reviewer: ```suggestion
| `action` | `application.authorized` | Reports that the seller has been linked to the application. |
```

> | `action` | `application.deauthorized` | Confirms that seller unlinked from application. |

Reviewer: ```suggestion
| `action` | `application.deauthorized` | Confirms that the seller has been unlinked from the application. |
```

> | `data.id`| `string`| ID of seller linked to application. |

Reviewer: ```suggestion
| `data.id`| `string`| ID of the seller linked to the application. |
```
